var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -37,7 +37,8 @@ class Kokkos(Package):
     variant('qthreads', default=False, description="enable Qthreads backend")
     variant('cuda', default=False, description="enable Cuda backend")
 
-    depends_on('hwloc')
+    # Specify that v1.x is required as v2.x has API changes
+    depends_on('hwloc@:1.11.9')
```

Comment:
I would use `hwloc@:1`. We don't know if a newer version of hwloc v1.x will be released.